homeassistant/components/hive/light.py

```diff
@@ -77,7 +77,13 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             get_new_color = kwargs[ATTR_HS_COLOR]
             hue = int(get_new_color[0])
             saturation = int(get_new_color[1])
-            new_color = (hue, saturation, 100)
+            brightness = int(
+                round(
+                    ((self.brightness if self.brightness is not None else 5) / 255)
```

Comment:
can we remove the inline if statement? It looks a bit messy here

Reply:
The inline if is OK IMHO, but I'm not convinced the change in this PR is fully correct.

Looking at the library, the `turnOn` method first sets the brightness if not `None`, then the color temperature if not None, and finally the color - which also includes brightness - if not None: https://github.com/Pyhass/Pyhiveapi/blob/b3ee963a8dea3a65eda2df0de5f3edfc8346a104/pyhiveapi/apyhiveapi/light.py#L411C1-L428C54

It's possible to call the turn_on service with the following combinations of brightness and color:
- Only a brightness change
- Only a color change - I believe this is the case addressed by this PR?
- Both a brightness and a color change - I don't think this case is handled by this PR?

I think the turn on service handler needs to be rewritten like this:

```py
new_brightness = None
new_color_temp = None
new_color = None

if ATTR_HS_COLOR in kwargs:
    # set new_color only, don't set new_brightness
    # do color calculation here, set brightness of the hsv-tuple (i.e. new_color) to current brightness
    if ATTR_BRIGHNESS in kwargs:
        # update brightness in the hsv-tuple (i.e. new_color) to the new brightness
elif ATTR_BRIGHNESS in kwargs:
    # set new_brightness
if ATTR_COLOR_TEMP in kwargs:
    # set new_color_temp
```